controllers/predictor_controller.go

```diff
@@ -213,32 +217,62 @@ func (pr *PredictorReconciler) ReconcilePredictor(ctx context.Context, nname typ
 	return ctrl.Result{}, nil
 }
 
+// validatePredictor checks if there are incompatibilities in the spec
+// Returns a string describing the reason a Predictor is invalid, empty if valid.
+func validatePredictor(predictor *api.Predictor) string {
+	// if it exists, inspect and validate the storage specification
+	if predictor.Spec.Storage == nil {
+		return ""
+	}
+	storage := predictor.Spec.Storage
+
+	if storage.Path != nil && predictor.Spec.Path != "" {
+		return "Only one of spec.path and spec.storage.path can be specified"
+	}
+
+	if storage.SchemaPath != nil && predictor.Spec.SchemaPath != nil {
+		return "Only one of spec.schemaPath and spec.storage.schemaPath can be specified"
+	}
+
+	// PersistentVolumeClaim is deprecated and was never supported
+	if storage.PersistentVolumeClaim != nil {
+		return "spec.storage.PersistentVolumeClaim is not supported"
```

Comment:
Quick question: Are we not going to support PVC for modelmesh? Just wondering why this spec was in the predictor previously. We may still want this feature when adding the new pullman library for single model serving.

Reply:
My understanding is that `storage.PersistentVolumeClaim` was added to the Predictor spec in the early development of ModelMesh serving, part of the initial draft of the Predictor type. The plan has always been to add support for loading models from PVCs, but S3 support was higher priority and we just never got to it.

Let me know if I can make the code clearer, but the purpose of my change here is to deprecate `storage.PersistentVolumeClaim` in favor of the new fields from the `StorageSpec` that do not have a key referencing the storage type (`storage.path`, `storage.parameters`, `storage.key`, etc.). PVCs are planned to be supported using the new storage spec and implementation with pullman.